apps/studio/src/components/ResourceSelector/ResourceSelector.tsx

```diff
@@ -1,157 +1,228 @@
-import { Suspense, useState } from "react"
-import {
-  Box,
-  Flex,
-  HStack,
-  Icon,
-  Skeleton,
-  Spacer,
-  Text,
-} from "@chakra-ui/react"
+import { Suspense, useEffect, useState } from "react"
+import { Box, Flex, HStack, Skeleton, Spacer, Text } from "@chakra-ui/react"
 import { Button, Link } from "@opengovsg/design-system-react"
-import { ResourceType } from "~prisma/generated/generatedEnums"
+import { ResourceType } from "@prisma/client"
 import { BiHomeAlt, BiLeftArrowAlt } from "react-icons/bi"
 
+import type { PendingMoveResource } from "~/features/editing-experience/types"
+import type { ResourceChildrenOfType } from "~/schemas/resource"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { sitePageSchema } from "~/pages/sites/[siteId]"
 import { trpc } from "~/utils/trpc"
 import { ResourceItem } from "./ResourceItem"
 
+interface ResourceSelectorProps {
+  onChange: (resourceId: string) => void
+  selectedResourceId?: string
+  existingResource?: PendingMoveResource | null
+  onlyShowFolders?: boolean
+}
+
+const generatePermalink = (parents: PendingMoveResource[]) => {
+  return parents.map((parent) => parent.permalink).join("/")
+}
+
 const SuspensableResourceSelector = ({
-  siteId,
-  selectedResourceId,
   onChange,
-  isDisabledFn,
+  selectedResourceId,
+  existingResource,
+  onlyShowFolders = false,
 }: ResourceSelectorProps) => {
-  const [ancestryStack] = trpc.resource.getAncestryOf.useSuspenseQuery({
-    siteId,
-    resourceId: selectedResourceId,
-  })
-  const [parentIdStack, setParentIdStack] = useState<string[]>(
-    ancestryStack.map((item) => item.id),
+  // NOTE: This is the stack of user's navigation through the resource tree
+  // NOTE: We should always start the stack from `/` (root)
+  // so that the user will see a full overview of their site structure
+  const [resourceStack, setResourceStack] = useState<PendingMoveResource[]>([])
+  const [isResourceHighlighted, setIsResourceHighlighted] =
```

Comment:
question(non-blocking): this `isResourceHighlighted` isn't very clear to me, is this referring to whether there is a resource that has been selected?

Reply:
yup!

I was thinking about this but `isResourceSelected` might be misleading too because it's unclear too selecting it (single click) and going into it (double click) means its selected either way

`isInFolder` (or equivalent) won't work for non-folder/collection pages

Reply:
lemme know if you have a better suggestion! i can't think of anything else 🥲

Reply:
> it's unclear too selecting it (single click) and going into it (double click) means its selected either way

Hmm I'm a bit confused. I thought this meant that there was a selection made and if the user were to insert the link at this point, we will insert the selection that the user has made. So if the user selected a normal page, this variable will be set to true. If the user selected a folder/collection, this variable will be set to true until the user double clicks, which then resets it to false.

Maybe I'm misunderstanding what this variable does? Might need a bit more clarification